streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -782,13 +810,27 @@ void to(final Serde<K> keySerde,
      *                    {@link WindowedStreamPartitioner} will be used&mdash;otherwise {@link DefaultPartitioner} will
      *                    be used
      * @param topic       the topic name
+     * @deprecated use {@code to(String, Produced}
      */
+    @Deprecated
     void to(final Serde<K> keySerde,
             final Serde<V> valSerde,
             final StreamPartitioner<? super K, ? super V> partitioner,
             final String topic);
 
     /**
+     * Materialize this stream to a topic using the provided {@link Produced} instance.
+     * The specified topic should be manually created before it is used (i.e., before the Kafka Streams application is
+     * started).
+     *
+     * @param produced    the options to use when producing to the topic;
+     *                    if not specified the default serde defined in the configs will be used
```

Comment:
This is a bit confusing since we do not allow `produced` to be null:

```
Objects.requireNonNull(produced, "Produced can't be null");
```

Or maybe we are just saying if the serdes are not provided within the `produced` object? If yes could you clarify this a bit more clearly?

Reply:
left over from a comment i'd copied - thanks